src/transport/FabricTable.cpp

```diff
@@ -350,21 +351,22 @@ CHIP_ERROR FabricInfo::VerifyCredentials(const ByteSpan & opCertArray, Validatio
     // It confirms that the certs link correctly (noc -> icac -> rcac), and have been correctly signed.
     ReturnErrorOnFailure(certificates.FindValidCert(nocSubjectDN, nocSubjectKeyId, context, &resultCert));
 
-    UncompressedPeerId uncompressedPeerId;
-    ReturnErrorOnFailure(ExtractPeerIdFromOpCert(certificates.GetCertSet()[nocCertIndex], &uncompressedPeerId));
+    RawPeerId rawPeerId;
+    ReturnErrorOnFailure(ExtractPeerIdFromOpCert(certificates.GetCertSet()[nocCertIndex], &rawPeerId));
 
     if (!icac.empty())
     {
-        FabricId icacFabric;
-        if (ExtractFabricIdFromCert(certificates.GetCertSet()[1], &icacFabric) == CHIP_NO_ERROR)
+        FabricId icacFabric = kUndefinedRawFabricId;
+        if (ExtractFabricIdFromCert(certificates.GetCertSet()[1], &icacFabric) == CHIP_NO_ERROR &&
+            icacFabric != kUndefinedRawFabricId)
         {
-            ReturnErrorCodeIf(icacFabric != uncompressedPeerId.GetUncompressedFabricId(), CHIP_ERROR_INVALID_CASE_PARAMETER);
+            ReturnErrorCodeIf(icacFabric != rawPeerId.GetRawFabricId(), CHIP_ERROR_INVALID_CASE_PARAMETER);
```

Comment:
Would be great if we had an error that made this abundantly clear, like `CHIP_ERROR_FABRIC_MISMATCH_ON_ICA`, which may occur in a variety of places (including UpdateNOC command handling)